### PR TITLE
Feature/graph with deletions

### LINKED ISSUE #1062: when creating a subgraph. then `.materialize` throws errors due to edge deletions

### LINKED ISSUE #1064: Exposoing edge deleitions to python

### PR DESCRIPTION
**Pull request by ljeub-pometry**
### What changes were proposed in this pull request?

- Rework of a lot of python wrappers
- Improved support for deletions 
- Exposed deletions to python
- Fixes some materialise issues (there seem to be more though)
- Introduces a prelude to simplify imports

### Why are the changes needed?

- Support different time semantics

### Does this PR introduce any user-facing change? If yes is this documented?

- Adds more classes, needs more documentation

### How was this patch tested?

added some deletion-specific tests (need to add more)

### Issues

closes #1064, hopefully fixes  #1062

### CODE REVIEW

File: raphtory/src/db/graph.rs [hunk 1]
```diff
@@ -60,38 +64,74 @@ pub struct InternalGraph {
...
+    fn from(value: InternalGraph) -> Self {
+        Self(Arc::new(value))
     }
 }
 
-impl WrappedGraph for Graph {
-    type Internal = InternalGraph;
-
-    fn graph(&self) -> &InternalGraph {
+// impl Deref for Graph {
```
Comment (miratepuffin) [inline raphtory/src/db/graph.rs:109]: Can remove

File: raphtory/src/db/graph_deletions.rs [hunk 2]
```diff
@@ -1,54 +1,154 @@
...
+
+impl<G: GraphViewOps> PartialEq<G> for GraphWithDeletions {
+    fn eq(&self, other: &G) -> bool {
+        graph_equal(self, other)
     }
 }
 
-impl<G: GraphViewOps> InheritCoreOps for GraphWithDeletions<G> {
-    type Internal = G;
+impl Inheritable for GraphWithDeletions {
```
Comment (miratepuffin) [inline raphtory/src/db/graph_deletions.rs:123]: Can we think of a better name for this?

**Review by miratepuffin — APPROVED** [review]

File: raphtory/src/db/graph_deletions.rs (hunk 2 above)
Comment (ljeub-pometry) [reply raphtory/src/db/graph_deletions.rs]: Renamed to `Base`

File: raphtory/src/db/graph.rs (hunk 1 above)
Comment (ljeub-pometry) [reply raphtory/src/db/graph.rs]: done

**Comment by ljeub-pometry** [comment]
Python wrappers are pretty clean now, so this should close #965

**Pull request merged**